src/main/java/org/ow2/proactive/catalog/service/BucketService.java

```diff
@@ -193,4 +194,18 @@ public void cleanAll() {
         bucketRepository.deleteAll();
     }
 
+    public BucketMetadata deleteEmptyBucket(long bucketId) {
+        BucketEntity bucket = bucketRepository.findBucketForUpdate(bucketId);
+
+        if (bucket == null) {
```

Comment:
i am not very sure, but when the bucketId does not exist here, the query will throw an exception instead of null maybe. Can you give just a manual test please. let me know the result please